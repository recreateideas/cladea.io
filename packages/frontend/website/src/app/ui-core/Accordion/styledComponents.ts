import styled, { css } from 'styled-components';
import {
    Accordion as MUIAccordion,
    AccordionDetails as MUIAccordionDetails,
    AccordionSummary as MUIAccordionSummary,
} from '@mui/material';

export const Container = styled.div``;
export const Accordion = styled(MUIAccordion)`
    ${(props) => {
        const {
            theme: {
                dsl: { palette },
            },
        } = props;
        return css`
            background-color: ${palette.neutrals.secondaryBg};
            box-shadow: none;
            background-image: none;
        `;
    }}
`;
export const AccordionDetails = styled(MUIAccordionDetails)`
    ${(props) => {
        const {
            theme: {
                dsl: {
                    layout: { padding },
                },
            },
        } = props;
        return css`
            padding: 0px ${padding.small} ${padding.small} ${padding.small};
        `;
    }}
`;
export const AccordionSummary = styled(MUIAccordionSummary)`
    ${(props) => {
        const {
            theme: {
                dsl: {
                    sizes,
                    layout: { padding },
                },
            },
        } = props;
        return css`
            &.Mui-expanded {
                min-height: ${sizes.accordion.medium};
            }
            > div {
                margin: ${padding.small} 0 !important;
            }
            padding: 0px ${padding.small};
        `;
    }}
`;
export const Title = styled.div``;
