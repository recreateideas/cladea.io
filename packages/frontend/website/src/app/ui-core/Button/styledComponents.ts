import styled, { css } from 'styled-components';
import { Button } from '@mui/material';

export const StyledButton = styled(Button)`
    ${(props) => {
        const {
            theme: {
                dsl: { palette },
            },
        } = props;
        return css`
            &.MuiButton-containedPrimary .MuiButton-label {
                color: ${palette.fonts.primaryInverted};
            }
        `;
    }}
`;
