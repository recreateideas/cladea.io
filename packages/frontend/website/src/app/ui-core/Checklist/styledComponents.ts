import styled from 'styled-components';

export const Container = styled.div`
    width: 100%;
    ul {
        background-color: inherit;
        .MuiListItemIcon-root {
            min-width: auto;
        }
        .MuiButtonBase-root {
        }
    }
`;
