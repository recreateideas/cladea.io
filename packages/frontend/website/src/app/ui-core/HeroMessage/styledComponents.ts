import styled, { css } from 'styled-components';

export const Container = styled.div<{ type: 'error' | 'warn' | 'success' }>`
    ${(props) => {
        const {
            theme: {
                dsl: { hexToRgba, styles, layout, palette },
            },
            type,
        } = props;
        const colorMap: {
            [type: string]: {
                color: string;
                backgroundColor: string;
            };
        } = {
            error: {
                color: palette.fonts.error,
                backgroundColor: hexToRgba(palette.secondary.orange[500], 10),
            },
            success: {
                color: palette.fonts.success,
                backgroundColor: hexToRgba(palette.secondary.green[500], 10),
            },
            warn: {
                color: palette.fonts.warn,
                backgroundColor: hexToRgba(palette.secondary.yellow[500], 10),
            },
        };
        return css`
            ${colorMap[type]}
            white-space: pre;
            padding: ${layout.padding.small};
            border-radius: ${styles.borderRadius};
            overflow-x: auto;
            .section {
                display: flex;
                .icon {
                    margin-right: ${layout.margin.small};
                }
            }
        `;
    }}
`;
