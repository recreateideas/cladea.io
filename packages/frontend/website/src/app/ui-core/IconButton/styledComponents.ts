import styled, { css } from 'styled-components';

export const Container = styled.div<{ shape?: 'square'; disabled?: boolean }>`
    ${(props) => {
        const {
            theme: {
                dsl: { palette, styles },
            },
            disabled,
            shape,
        } = props;
        const bordersRadius = shape === 'square' ? styles.borderRadius : undefined;
        return css`
            ${bordersRadius &&
            css`
                .MuiButtonBase-root {
                    border-radius: ${bordersRadius};
                }
            `}
            ${disabled &&
            css`
                &,
                * {
                    cursor: no-drop;
                    svg {
                        fill: ${palette.neutrals.inactive} !important;
                    }
                }
            `}
        `;
    }}
`;
