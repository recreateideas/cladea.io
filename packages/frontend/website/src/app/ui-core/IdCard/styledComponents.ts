import styled, { css } from "styled-components";
import { avatar } from "../images";

export const Container = styled.div<{ breakpoint: string }>`
  ${(props) => {
    const {
      theme: {
        dsl: { layout, palette },
      },
      breakpoint,
    } = props;
    const isSmall = breakpoint === "sm";
    const width = isSmall ? "100%" : "30vw";
    return css`
      border-radius: 12px;
      min-width: ${width};
      background-color: ${palette.neutrals.quaternaryBg};
      border: solid 1px ${palette.neutrals.borders};
      overflow: hidden;
      .background {
        width: 100%;
        height: 60vh;
        fill: ${palette.neutrals.secondaryBg};
      }
      z-index: ${layout.namedZIndex.leftSidebar - 1};
      .bio {
        display: flex;
        flex-direction: column;
      }
      .avatar {
        position: relative;
        border-radius: 100%;
        overflow: hidden;
        background: ${palette.neutrals.primaryBg};
        width: 20vh;
        height: 20vh;
        margin: 24px auto;
        .image-container {
          border-radius: 100%;
          overflow: hidden;
          width: inherit;
          height: inherit;
          background-image: url(${avatar});
          background-position: center;
          background-repeat: no-repeat;
          background-size: cover;
        }
        .space {
          background: unset !important;
        }
      }
      .name {
        display: flex;
        letter-spacing: 1px;
        justify-content: center;
        gap: 16px;
        .first {
        }
        .last {
          color: ${palette.tertiary.pink[500]};
        }
      }
      .position {
        width: 100%;
        .title {
          color: ${palette.secondary.neon[500]};
          text-align: center;
        }
      }
      .contacts {
        display: flex;
        flex-direction: column;
        gap: 16px;
        margin: ${isSmall ? "2vh" : "24px"} 24px;
        padding: ${isSmall ? "2vh" : "24px"};
        border: solid 1px ${palette.neutrals.borders};
        border-radius: 8px;
        background-color: ${palette.neutrals.secondaryBg};
        .contact-container {
          width: 100%;
          /* height: 40px; */
          display: flex;
          justify-content: center;
          .href-container {
            text-decoration: none;
            display: flex;
            gap: 16px;
            color: ${palette.fonts.primary};
            cursor: pointer;
            img {
              width: 36px;
            }
            .href-text {
              display: flex;
              flex-direction: column;
              justify-content: center;
              width: 80px;
            }
          }
        }
      }
    `;
  }};
`;

export const ToastContainer = styled.div`
  ${(props) => {
    const {
      theme: {
        dsl: { palette },
      },
    } = props;
    return css`
      .toast {
        display: block;
        position: fixed;
        top: 50%;
        left: 50%;
        transform: translateX(-50%);
        min-width: 450px;
        background-color: ${palette.neutrals.quaternaryBg};
        box-shadow: ${palette.neutrals.boxShadow(palette.secondary.neon[500])};
        border: solid 1px ${palette.neutrals.borders};
        .btn-close {
          color: ${palette.fonts.primary};
          background-color: ${palette.secondary.neon[500]};
        }
        .toast-header {
          justify-content: space-between;
          background-color: ${palette.neutrals.primaryBg};
          button {
            margin-right: 0;
          }
          .header-title {
            font-weight: 600;
            width: 100%;
            color: ${palette.secondary.neon[500]};
          }
        }

        .body-content {
          width: 100%;
          display: flex;
          justify-content: space-between;
          .copied-value {
            color: ${palette.tertiary.pink[500]};
          }
        }
      }
    `;
  }};
`;
