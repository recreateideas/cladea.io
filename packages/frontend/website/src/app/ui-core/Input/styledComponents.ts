import styled from 'styled-components';
import { CustomCss } from '@types';

export const Container = styled.div<{ customCss?: CustomCss }>`
    width: 100%;
    display: flex;
    flex-direction: column;
    justify-content: center;
    ${(props) => props.customCss}
`;
