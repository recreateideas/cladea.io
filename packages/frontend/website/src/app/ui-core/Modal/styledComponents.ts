import styled, { css } from 'styled-components';
import { Modal } from 'react-bootstrap';

const StyledModal = styled(Modal)`
    ${(props) => {
        const {
            theme: {
                dsl: { palette, layout },
            },
        } = props;
        return css`
            .modal-content {
                box-shadow: ${palette.neutrals.boxShadow()};
                border: solid 1px ${palette.neutrals.borders};
            }
            .modal-body {
                max-height: 500px;
                overflow: auto;
            }
            .modal-content,
            .modal-header,
            .modal-footer {
                background-color: ${palette.neutrals.primaryBg};
            }
            .modal-body,
            .modal-header,
            .modal-footer {
                border: none;
                padding: ${layout.padding.small};
            }
            .close {
                color: ${palette.fonts.primary};
            }
        `;
    }}
`;

export { StyledModal };
