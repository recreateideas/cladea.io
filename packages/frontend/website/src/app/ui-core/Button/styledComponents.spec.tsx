import { StyledButton } from './styledComponents';
import { TestProvider } from 'src/tests';

const { renderWithProviderAndTheme } = TestProvider({});

describe('Button - styledComponents', () => {
    describe('<StyledButton/>', () => {
        it('should render from left', () => {
            const root = renderWithProviderAndTheme(<StyledButton />);
            expect(root).toMatchSnapshot();
        });
    });
});
