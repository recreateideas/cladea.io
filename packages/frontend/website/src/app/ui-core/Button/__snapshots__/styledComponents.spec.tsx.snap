// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button - styledComponents <StyledButton/> should render from left 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
}

.c0::-moz-focus-inner {
  border-style: none;
}

.c0.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

.c1 {
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 500;
  font-size: 0.875rem;
  line-height: 1.75;
  -webkit-letter-spacing: 0.02857em;
  -moz-letter-spacing: 0.02857em;
  -ms-letter-spacing: 0.02857em;
  letter-spacing: 0.02857em;
  text-transform: uppercase;
  min-width: 64px;
  padding: 6px 8px;
  border-radius: 4px;
  -webkit-transition: background-color 250ms cubic-bezier(0.4,0,0.2,1) 0ms,box-shadow 250ms cubic-bezier(0.4,0,0.2,1) 0ms,border-color 250ms cubic-bezier(0.4,0,0.2,1) 0ms,color 250ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: background-color 250ms cubic-bezier(0.4,0,0.2,1) 0ms,box-shadow 250ms cubic-bezier(0.4,0,0.2,1) 0ms,border-color 250ms cubic-bezier(0.4,0,0.2,1) 0ms,color 250ms cubic-bezier(0.4,0,0.2,1) 0ms;
  color: #12D8F7;
}

.c1:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: rgba(18,216,247,0.04);
}

.c1.Mui-disabled {
  color: rgba(0,0,0,0.26);
}

.c2.MuiButton-containedPrimary .MuiButton-label {
  color: #ffffff;
}

@media print {
  .c0 {
    color-adjust: exact;
  }
}

@media (hover:none) {
  .c1:hover {
    background-color: transparent;
  }
}

<button
  class="c0 MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root c1 c2"
  tabindex="0"
  type="button"
/>
`;
